internal/cli/certificate.go

```diff
@@ -0,0 +1,367 @@
+package cli
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	apiclient "github.com/flightctl/flightctl/internal/api/client"
+	"github.com/flightctl/flightctl/internal/client"
+	fccrypto "github.com/flightctl/flightctl/internal/crypto"
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"golang.org/x/term"
+	"sigs.k8s.io/yaml"
+)
+
+const secondsInDay = 86400
+
+type CertificateOptions struct {
+	GlobalOptions
+	Name            string
+	SignerName      string
+	Expiration      string
+	OutputFormat    string
+	PrivKeyPassword string
+}
+
+func DefaultCertificateOptions() *CertificateOptions {
+	return &CertificateOptions{
+		GlobalOptions: DefaultGlobalOptions(),
+		SignerName:    "ca",
+		OutputFormat:  "components",
+	}
+}
+
+func NewCmdCertificate() *cobra.Command {
+	o := DefaultCertificateOptions()
+	cmd := &cobra.Command{
+		Use: "certificate [subcommand]",
+		// more subcommands may be added later
+		Short:     "request a new enrollment certificate for a device with 'certificate request'",
+		Args:      cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+		ValidArgs: []string{"request"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			return o.Run(cmd.Context(), args)
+		},
+		SilenceUsage: true,
+	}
+	o.Bind(cmd.Flags())
+	return cmd
+}
+
+func (o *CertificateOptions) Bind(fs *pflag.FlagSet) {
+	o.GlobalOptions.Bind(fs)
+	fs.StringVarP(&o.Name, "name", "n", "mycsr", "specify a name for the certificate signing request")
+	fs.StringVarP(&o.SignerName, "signer", "s", "ca", "specify certificate signer - options currently supported: 'ca'")
```

Comment:
Can we replace "ca" here immediately, so it doesn't bite us later? `enrollment`, `enrollmentSigner` or similar?

Reply:
I was going to leave this to be included in @kot-begemot-uk 's PR but I am fine to change it here.